formal-spec/measurements.tex

```diff
@@ -174,6 +177,102 @@ \section{Impact on performance}
 %% the section above.
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 
+\subsection{Processing Time and Memory Consumption}
+Clearly the most processing intesive task of the update mechanism is the tally phase. It is the phase where all the collected votes are counted in order to reach at a decision for a specific proposal. 
+
+We start with a theoretical time complexity analysis where we assume a worst-case scenario, where we have $n$ participants that all of them vote by submiting a single vote. Also we assume that we have a single proposal, so that within a voting period, the number $n$ of participants coincides to the number of submitted ballots.
+
+In the following we try to break up the operations during the tally phase. In the heart of the tally phase lies the following function call, which is called three times sequencially for each proposal.
```

Comment:
I wouldn't mention that it is called three times. We implemented this in this way, but we will likely change this to make a single pass.

Reply:
OK